### PR TITLE
Fix/issue 2824 - Making sure the data is string before parsing into JSON in order edit page

### LINKED ISSUE #2824: Console Error When Configuring Product Add-ons with WooCommerce Automated Taxes Enabled
**Bug Report**

**Describe the bug**  
When WooCommerce Shipping and Tax is activated with "Automated tax" enabled, configuring product addons during manual order creation throws a JavaScript error in the console.

**Console Error**  
```
Uncaught TypeError: r.data.indexOf is not a function  
File: woocommerce-services-new-order-taxjar-2.8.2.js  
Line: if (r.data && r.data.indexOf("woocommerce_calc_line_taxes") > 0)
```

**To Reproduce**  
1. Install WooCommerce Product Add-ons and configure a global addon.
2. Go to **WooCommerce > Settings > Tax** and enable "Automated taxes."
3. Go to **WooCommerce > Orders > Add New**, add a line item, then click on the "Configure" button.
4. Fill in the addon field in the popup, click "Done," and observe the console error.

**Expected behavior**  
The product addon should be configured without errors, and the order process should continue without getting stuck on the loading icon.

**Screenshots**  

https://d.pr/i/RAW1tf 
Full Size: https://d.pr/i/RAW1tf

### PR DESCRIPTION
**Pull request by iyut**
## Description
Making sure the data is string before parsing into JSON in order edit page. It needs to be done this way to make sure that the `indexOf()` function is not called.

### Related issue(s)
- Fixes #2824 

### Steps to reproduce & screenshots/GIFs
1. Install [Product Add-Ons plugin](https://woocommerce.com/products/product-add-ons/) is installed.
2. Create one product add-ons.
3. Go to **WooCommerce > Settings > Tax** and enable "Automated taxes."
4. Go to **WooCommerce > Orders > Add New**, add a line item, then click on the "Configure" button.
image

5. Fill in the addon field in the popup, click "Done".
6. The product add ons will be added successfully.

### Checklist

- [ ] unit tests

- [x] `changelog.txt` entry added
- [x] `readme.txt` entry added

### CODE REVIEW

**Review by samnajian — APPROVED** [review]

File: changelog.txt [hunk 1]
```diff
@@ -1,5 +1,8 @@
 *** WooCommerce Shipping & Tax Changelog ***
 
+= 2.8.5 - 2024-xx-xx =
+* Fix   - Adding Product Add-Ons on order edit page is stucked when automated tax is enabled.
```
Comment (dustinparker) [inline changelog.txt]:
```suggestion
* Fix   - Adding Product Add-Ons on order edit page is broken when automated tax is enabled.
```

File: client/new-order-taxjar.js [hunk 2]
```diff
@@ -11,7 +11,7 @@ jQuery( document ).ready( function() {
 	const TaxJarOrder = function( $ ){ // eslint-disable-line no-unused-vars
 		$( document ).ajaxSend( function( event, request, settings ) {
 			// Making sure it's a recalculate event by checking `woocommerce_calc_line_taxes` before parsing to JSON.
-			if ( settings.data && settings.data.indexOf( 'woocommerce_calc_line_taxes' ) > 0 ) {
+			if ( settings.data && typeof settings.data === 'string' && settings.data.indexOf( 'woocommerce_calc_line_taxes' ) > 0 ) {
```
Comment (dustinparker) [inline client/new-order-taxjar.js:14]:
```suggestion
			if ( settings.data && 'string' === typeof settings.data && settings.data.indexOf( 'woocommerce_calc_line_taxes' ) > 0 ) {
```

File: readme.txt [hunk 3]
```diff
@@ -81,6 +81,9 @@ The source code is freely available [in GitHub](https://github.com/Automattic/wo
 
 == Changelog ==
 
+= 2.8.5 - 2024-xx-xx =
+* Fix   - Adding Product Add-Ons on order edit page is stucked when automated tax is enabled.
```
Comment (dustinparker) [inline readme.txt]:
```suggestion
* Fix   - Adding Product Add-Ons on order edit page is broken when automated tax is enabled.
```

**Review by dustinparker — APPROVED** [review]
That fixed the issue! Just left a few suggestions.

File: changelog.txt (hunk 1 above)
Comment (PanosSynetos) [reply changelog.txt]: This is not strictly related to product add-ons. As mentioned in the issue

> Product Bundles uses a FormData object to set the ajax data, and not a generic object or array. FormData doesn’t have an indexOf method. The same issue exists with Product Add-Ons and Composite Products and [edited] any other plugin that doesn’t use an array for the data


So, maybe something more generic?

```
* Fix   - Fixed an issue that prevented editing an order when automated tax is enabled.
```

File: readme.txt (hunk 3 above)
Comment (PanosSynetos) [reply readme.txt]: Same as my [comment](https://github.com/Automattic/woocommerce-services/pull/2829#discussion_r1878151147) above

File: changelog.txt (hunk 1 above)
Comment (iyut) [reply changelog.txt]: Thanks for the suggestion @PanosSynetos , I'll change the wording with your suggestion.

**Pull request merged**